cpp/src/groupby/sort/sort_helper.cu

```diff
@@ -226,19 +227,11 @@ sort_groupby_helper::index_vector const& sort_groupby_helper::group_labels(
 
   if (num_keys(stream) == 0) return group_labels;
 
-  thrust::uninitialized_fill(rmm::exec_policy(stream),
-                             group_labels.begin(),
-                             group_labels.end(),
-                             index_vector::value_type{0});
-  thrust::scatter(rmm::exec_policy(stream),
-                  thrust::make_constant_iterator(1, decltype(num_groups(stream))(1)),
-                  thrust::make_constant_iterator(1, num_groups(stream)),
-                  group_offsets(stream).begin() + 1,
-                  group_labels.begin());
-
-  thrust::inclusive_scan(
-    rmm::exec_policy(stream), group_labels.begin(), group_labels.end(), group_labels.begin());
-
+  cudf::detail::fill_segmented_labels(group_offsets(stream).begin(),
+                                      group_offsets(stream).end(),
+                                      group_labels.begin(),
+                                      group_labels.end(),
+                                      stream);
```

Comment:
I'm curious if a vector search solution is faster than scatter + scan solution. Can I get a microbench please? Doesn't have to be of libcudf APIs, can bench thrust calls.

Reply:
I did that: (almost) no difference at all. Maybe because these calls are too small.

Reply:
I agree about the perf. The binary search approach would be N*logn work whereas scatter+scan is N, but with 2 kernels instead of 1. It'd be a bit of a toss up on which will perform better in the cases we care about. I suspect the binary search approach will be better for small input sizes, but scatter/scan better for large.

I'd also like to see some benchmarks across a few input sizes.

Reply:
Does the time here include sorting (pre-sorted == false)?

Reply:
The benchmarks are for structs column min groupby. The aggregations with values columns having structs type are always processed by sorting.

Reply:
I'd write a simple standalone benchmark of just comparing the raw thrust algorithm calls to avoid including any overhead that obscures the differences between the two approaches.